modules/widgets/mailchimp.php

```diff
@@ -32,28 +52,119 @@ function __construct() {
+				$instance,
+				array(
+					'code'      => '',
+					'cssClass'  => '',
+					'popupMode' => '',
+					'delay'     => '0',
+				)
+			);
+
+			if ( ! empty( $instance['code'] ) ) {
+				// Regular expresion that will match maichimp shortcode.
+				$regex = '(\[mailchimp_subscriber_popup[^\]]+\])';
+
+				// Check if the shortcode exists.
+				preg_match( $regex, $instance['code'], $matches );
+
+				// Process the shortcode only, if exists.
+				if ( ! empty( $matches[0] ) ) {
+					echo do_shortcode( $matches[0] );
+				}
+			} else {
+				$instance['interests'] = empty( $instance['interests'] ) ? array() : explode( '_', $instance['interests'] );
 
-			// Regular expresion that will match maichimp shortcode.
-			$regex = '(\[mailchimp_subscriber_popup[^\]]+\])';
+				$is_amp_request = class_exists( 'Jetpack_AMP_Support' ) && Jetpack_AMP_Support::is_amp_request();
 
-			// Check if the shortcode exists.
-			preg_match( $regex, $instance['code'], $matches );
+				$popup_mode = 'on' === $instance['popupMode'];
+				// Use the same outpu as the block.
+				$output       = render_block( $instance );
+				$form_classes = 'jetpack-mailchimp-widget-form';
 
-			// Process the shortcode only, if exists.
-			if ( ! empty( $matches[0] ) ) {
-				echo do_shortcode( $matches[0] );
+				if ( ! empty( $instance['cssClass'] ) ) {
+					$form_classes .= ' ' . $instance['cssClass'];
+				}
+
+				if ( $popup_mode ) {
+
+					if ( $is_amp_request ) {
+
+						wp_enqueue_style(
+							'jetpack-mailchimp-popup-style',
+							Assets::get_file_url_for_environment(
+								'_inc/build/widgets/mailchimp/css/popup-amp.min.css',
+								'modules/widgets/mailchimp/css/popup-amp.css'
+							),
+							array(),
+							'20200618'
+						);
+
+						$output = sprintf(
+							'<amp-user-notification class="%2$s-wrapper" id="%2$s" layout="nodisplay">
+								<div class="%1$s">
+									<button class="%2$s-close" on="tap:%2$s.dismiss"></button>
+									%3$s
+								</div>
+							</amp-user-notification>',
+							$form_classes,
+							'jetpack-mailchimp-widget',
+							$output
+						);
+					} else {
+
+						wp_enqueue_script(
+							'jetpack-mailchimp-popup',
+							Assets::get_file_url_for_environment(
+								'_inc/build/widgets/mailchimp/js/popup.min.js',
+								'modules/widgets/mailchimp/js/popup.js'
+							),
+							array( 'jquery' ),
+							'20200615',
+							true
+						);
+
+						wp_localize_script(
+							'jetpack-mailchimp-popup',
+							'jetpackMailchimpPopup',
+							array(
+								'delay' => $instance['delay'],
+							)
+						);
+
+						wp_enqueue_style(
+							'jetpack-mailchimp-popup-style',
+							Assets::get_file_url_for_environment(
+								'_inc/build/widgets/mailchimp/css/popup.min.css',
+								'modules/widgets/mailchimp/css/popup.css'
+							),
+							array(),
+							'20200615'
+						);
+
+						$output = sprintf(
+							'<div class="%s" style="display:none;">%s</div>',
+							$form_classes,
```

Comment:
We may want to escape things with `esc_attr` here.

Reply:
@jeherve I added a call for `esc_attr` in the last commit, thanks!